File: ios/Classes/FlutterLibjeidPluginStreamHandler.swift
import Foundation
import Flutter

class FlutterLibjeidPluginEventStreamHandler: NSObject, FlutterStreamHandler {
    var eventSink: FlutterEventSink?
    
    func emit(event: FlutterLibjeidEvent) {
        DispatchQueue.main.async {
            self.eventSink?(event.toJSON());
        }
    }

    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil

        return nil
    }
}

File: android/src/main/java/ponos/tech/flutter_libjeid/FlutterLibjeidPluginStreamHandler.kt
package ponos.tech.flutter_libjeid

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.EventChannel.StreamHandler


class FlutterLibjeidPluginStreamHandler : StreamHandler {
    private val mainHandler: Handler = Handler(Looper.getMainLooper())
    private var eventSink: EventChannel.EventSink? = null

    fun emit(event: FlutterLibjeidEvent) {
        val runnable = Runnable { eventSink?.success(event.toJSON()) }
        mainHandler.post(runnable)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}